### PR TITLE
Forward port blake2 storage support

### PR DESCRIPTION
**Pull request by bkchr**
https://github.com/paritytech/substrate/pull/2268

### CODE REVIEW

File: core/executor/src/wasm_executor.rs [hunk 1]
```diff
@@ -940,6 +979,21 @@ mod tests {
...
+			WasmExecutor::new().call(&mut ext, 8, &test_code[..], "test_blake2_128", &[]).unwrap(),
+			blake2_128(&b""[..]).encode()
+		);
+		assert_eq!(
+			WasmExecutor::new().call(&mut ext, 8, &test_code[..], "test_blake2_128", b"Hello world!").unwrap(),
+			blake2_128(&b"Hello world!"[..]).encode()
+		);
+	}
+
+
```
Comment (thiolliere) [inline core/executor/src/wasm_executor.rs]:
```suggestion
```

also maybe we could add test twox_64_should_work.

File: core/executor/wasm/src/lib.rs [hunk 2]
```diff
@@ -68,6 +68,7 @@ impl_stubs!(
 		input.to_vec()
 	},
 	test_blake2_256 => |input| blake2_256(input).to_vec(),
+	test_blake2_128 => |input| blake2_128(input).to_vec(),
```
Comment (thiolliere) [inline core/executor/wasm/src/lib.rs]: did I forget test_twox_64 as well ?

File: srml/support/procedural/tools/src/syn_ext.rs [hunk 3]
```diff
@@ -72,7 +72,7 @@ groups_impl!(Braces, Brace, Brace, parse_braces);
 groups_impl!(Brackets, Bracket, Bracket, parse_brackets);
 groups_impl!(Parens, Paren, Parenthesis, parse_parens);
 
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```
Comment (thiolliere) [inline srml/support/procedural/tools/src/syn_ext.rs]: hmm I now remember I introduced this Clone before creating the new enum HasherKind. So I very suspect this change is not needed anymore.

Comment (bkchr) [reply srml/support/procedural/tools/src/syn_ext.rs]: As we will remove this anyway, it is no problem.

**Pull request merged**